src/test/shell/integration/watchfs_test.sh

```diff
@@ -99,4 +99,49 @@ function test_both() {
   expect_not_log "VFS stat.*${pkg}/whocares.in"
 }
 
+function test_special_chars() {
+  local -r subdir="${FUNCNAME[0]} bazel 🌱"
+  local -r workspace="$TEST_TMPDIR/$subdir/workspace"
+  mkdir -p "$workspace" || fail "mkdir -p $workspace"
+
+  cd "$workspace" || fail "cd $workspace"
+  setup_module_dot_bazel
+
+  touch BUILD
+  bazel info
+  bazel build --watchfs //... &> "$TEST_log" || fail "Expected success."
+  expect_not_log "Hello, Unicode!"
+
+  mkdir pkg || fail "mkdir pkg"
+  cat > pkg/BUILD << 'EOF'
+print("Hello, Unicode!")
+
+genrule(
+    name = "foo",
+    srcs = ["foo 🌱.in"],
```

Comment:
This doesn't work at the HEAD? According to my occasional poking on Linux, it does, but I didn't check any other operating systems.

Reply:
Nevermind. This is in `watchfs_test`.